packages/react/dropdown-menu/src/DropdownMenu.tsx

```diff
@@ -146,37 +182,49 @@ const DropdownMenuContent = React.forwardRef((props, forwardedRef) => {
   } = props;
   const context = useDropdownMenuContext(CONTENT_NAME);
   return (
-    <SubmenuContext.Provider value={true}>
-      <MenuPrimitive.Content
-        id={context.contentId}
-        {...contentProps}
-        ref={forwardedRef}
-        disableOutsidePointerEvents={disableOutsidePointerEvents}
-        disableOutsideScroll={disableOutsideScroll}
-        portalled={portalled}
-        style={{
-          ...props.style,
-          // re-namespace exposed content custom property
-          ['--radix-dropdown-menu-content-transform-origin' as any]: 'var(--radix-popper-transform-origin)',
-        }}
-        trapFocus
-        onCloseAutoFocus={composeEventHandlers(onCloseAutoFocus, (event) => {
-          event.preventDefault();
-          context.triggerRef.current?.focus();
-        })}
-        onPointerDownOutside={composeEventHandlers(
-          props.onPointerDownOutside,
-          (event) => {
-            const target = event.target as HTMLElement;
-            const targetIsTrigger = context.triggerRef.current?.contains(target);
-            // prevent dismissing when clicking the trigger
-            // as it's already setup to close, otherwise it would close and immediately open.
-            if (targetIsTrigger) event.preventDefault();
-          },
-          { checkForDefaultPrevented: false }
-        )}
-      />
-    </SubmenuContext.Provider>
+    <ContentContext.Provider value={true}>
+      {context.isSubmenu ? (
+        <MenuPrimitive.Content
+          {...contentProps}
+          ref={forwardedRef}
+          disableOutsidePointerEvents={disableOutsidePointerEvents}
+          disableOutsideScroll={disableOutsideScroll}
+          portalled={portalled}
+          onCloseAutoFocus={onCloseAutoFocus}
```

Comment:
As we're now being explicit with the submenu content distinction I don't think we need to pass these anymore.

Reply:
Yeah looks like it was never needed to be honest, given they would be part of `...contentProps`

Reply:
I had thought the same thing initially but then we would be spreading the knowledge of how submenus should behave across two different components. 

My thinking here was that we would just pass them along and leave it to `Menu` to determine how to handle them since it already decides what to do with these in the context of a submenu.

Given both of your feedback, I think I need to change the boolean from `context.isSubmenu` to `context.isRootMenu`. Because, `DropdownMenu` doesn't actually need to know if it's a submenu or not, it only need to know if its a root menu to handle `DropdowMenu` specific things slightly differently, otherwise it just passes everything through to `Menu` to handle.

Reply:
@benoitgrelard @andy-hook thoughts on this one?

Reply:
Oh I see I didn't open it up higher so it's more to do with the default values right? that's why they end up being passed explicitly. That makes sense to me then.

Reply:
I much prefer this 👍